kernel/src/actions/mod.rs

```diff
@@ -113,7 +145,7 @@ pub struct Add {
     pub stats: Option<String>,
 
     /// Map containing metadata about this logical file.
-    pub tags: HashMap<String, Option<String>>,
+    pub tags: Option<HashMap<String, Option<String>>>,
```

Comment:
tags are optional in the spec, so this is a bug fix

Reply:
Can we please just remove the `Option<String>` business? It complicates the schema handling and there's nothing in the Delta spec that allows for it -- add.tags is a string-string map.

Reply:
(ah, it's being tracked in the other comment... but still I wonder why wait)

Reply:
Indeed. Changed to <String,String> everywhere (seems to work fine)